### PR TITLE
Refactor instructions part1

### PR DESCRIPTION
**Pull request by sjillen**
- refactor instructions form 
- refactor Solend and Raydium itxs
- remove mango instructions

### CODE REVIEW

File: hooks/useInstructionFormBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
+import yup from 'yup'
+import { isFormValid } from '@utils/formValidation'
```
Comment (0xCryptoSheik) [inline hooks/useInstructionFormBuilder.ts]: internal imports after 3rd party imports pls :pray:

File: pages/dao/[symbol]/proposal/components/instructions/ProposalForm.tsx [hunk 2]
```diff
@@ -0,0 +1,203 @@
+import useGovernedMultiTypeAccounts from '@hooks/useGovernedMultiTypeAccounts'
+import { ProgramAccount, Governance } from '@solana/spl-governance'
+import { GovernedMultiTypeAccount } from '@utils/tokens'
+import { useState } from 'react'
```
Comment (0xCryptoSheik) [inline pages/dao/[symbol]/proposal/components/instructions/ProposalForm.tsx]: imports

File: pages/dao/[symbol]/proposal/components/instructions/ProposalForm.tsx [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+        return <Clawback index={index} governance={governance} />
+      default:
+        null
+    }
+  }
+
+  return (
+    <>
+      {![
+        Instructions.Transfer,
```
Comment (0xCryptoSheik) [inline pages/dao/[symbol]/proposal/components/instructions/ProposalForm.tsx:31]: move this list to its own variable in another file ?

File: pages/dao/[symbol]/proposal/components/instructions/ProposalForm.tsx [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+}) => {
+  const { governedMultiTypeAccounts } = useGovernedMultiTypeAccounts()
+
+  const shouldBeGoverned = index !== 0 && governance
+
+  const [governedAccount, setGovernanceAccount] = useState<
+    GovernedMultiTypeAccount | undefined
+  >()
+
+  const getCurrentInstruction = () => {
```
Comment (0xCryptoSheik) [inline pages/dao/[symbol]/proposal/components/instructions/ProposalForm.tsx]: move this function to another file ?

File: pages/dao/[symbol]/proposal/components/instructions/Raydium/AddLiquidityToPool.tsx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+          <Select
+            label="Slippage (%)"
+            value={form.slippage}
+            onChange={(value) =>
+              handleSetForm({ value, propertyName: 'slippage' })
+            }
+            error={formErrors['slippage']}
+          >
+            <SelectOptionList list={[0.5, 1, 2]} />
```
Comment (0xCryptoSheik) [inline pages/dao/[symbol]/proposal/components/instructions/Raydium/AddLiquidityToPool.tsx]: put available values in their own variable at top of file like `SLIPPAGE_OPTIONS = [...]`

File: pages/dao/[symbol]/proposal/components/instructions/Raydium/AddLiquidityToPool.tsx [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+          <Select
+            label="Fixed Side"
+            value={form.fixedSide}
+            placeholder="Please select..."
+            onChange={(value) =>
+              handleSetForm({ value, propertyName: 'fixedSide' })
+            }
+            error={formErrors['fixedSide']}
+          >
+            <SelectOptionList list={['base', 'quote']} />
```
Comment (0xCryptoSheik) [inline pages/dao/[symbol]/proposal/components/instructions/Raydium/AddLiquidityToPool.tsx]: put in own variable

File: pages/dao/[symbol]/proposal/components/instructions/Raydium/RemoveLiquidityFromPool.tsx [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+      <Select
+        label="Raydium Liquidity Pool"
+        value={form.liquidityPool}
+        placeholder="Please select..."
+        onChange={(value) =>
+          handleSetForm({ value, propertyName: 'liquidityPool' })
+        }
+        error={formErrors['liquidityPool']}
+      >
+        <SelectOptionList list={Object.keys(liquidityPoolKeysList)} />
```
Comment (0xCryptoSheik) [inline pages/dao/[symbol]/proposal/components/instructions/Raydium/RemoveLiquidityFromPool.tsx]: move to its own variable outside of the component, always try to have stable references when possible (here the list is re-computed at each render, which doesn't really matter here but still)

File: pages/dao/[symbol]/proposal/components/instructions/Solend/RefreshObligation.tsx [hunk 8]
```diff
@@ -1,146 +1,63 @@
...
-      </Select>
-    </>
+    <Select
+      label="Token Name to refresh obligation for"
+      value={form.mintName}
+      placeholder="Please select..."
+      onChange={(value) => handleSetForm({ value, propertyName: 'mintName' })}
+      error={formErrors['baseTokenName']}
+    >
+      <SelectOptionList list={SolendConfiguration.getSupportedMintNames()} />
```
Comment (0xCryptoSheik) [inline pages/dao/[symbol]/proposal/components/instructions/Solend/RefreshObligation.tsx]: move outside of the component render and place it in own variable

`const SOLEND_OPTIONS = ...` or something

File: pages/dao/[symbol]/proposal/components/instructions/Solend/RefreshReserve.tsx [hunk 9]
```diff
@@ -1,144 +1,65 @@
...
-      </Select>
-    </>
+    <Select
+      label="Token Name to refresh reserve for"
+      value={form.mintName}
+      placeholder="Please select..."
+      onChange={(value) => handleSetForm({ value, propertyName: 'mintName' })}
+      error={formErrors['baseTokenName']}
+    >
+      {SolendConfiguration.getSupportedMintNames().map((value) => (
```
Comment (0xCryptoSheik) [inline pages/dao/[symbol]/proposal/components/instructions/Solend/RefreshReserve.tsx]: `SolendConfiguration.getSupportedMintNames()` in own variable outside of render

Comment (0xCryptoSheik) [reply pages/dao/[symbol]/proposal/components/instructions/Solend/RefreshReserve.tsx]: Also don't you have another component for this  which takes the list as props ?

File: pages/dao/[symbol]/proposal/components/instructions/Solend/WithdrawObligationCollateralAndRedeemReserveLiquidity.tsx [hunk 10]
```diff
@@ -1,172 +1,88 @@
...
         placeholder="Please select..."
         onChange={(value) => handleSetForm({ value, propertyName: 'mintName' })}
         error={formErrors['baseTokenName']}
       >
-        {SolendConfiguration.getSupportedMintNames().map((value) => (
-          <Select.Option key={value} value={value}>
-            {value}
-          </Select.Option>
-        ))}
+        <SelectOptionList list={SolendConfiguration.getSupportedMintNames()} />
```
Comment (0xCryptoSheik) [inline pages/dao/[symbol]/proposal/components/instructions/Solend/WithdrawObligationCollateralAndRedeemReserveLiquidity.tsx]: variable outside of render

**Comment by 0xCryptoSheik** [comment]
:+1: great progress, only a few nitpick comments on my part

**Review by 0xCryptoSheik — APPROVED** [review]

File: pages/dao/[symbol]/proposal/components/instructions/ProposalForm.tsx (hunk 3 above)
Comment (sjillen) [reply pages/dao/[symbol]/proposal/components/instructions/ProposalForm.tsx:31]: I would prefer to keep that here.
The objective being to make it disappear over time, I find it a nice incentive to have that ugly list screaming at your face everytime you open that file

File: pages/dao/[symbol]/proposal/components/instructions/Solend/RefreshReserve.tsx (hunk 9 above)
Comment (sjillen) [reply pages/dao/[symbol]/proposal/components/instructions/Solend/RefreshReserve.tsx]: yes indeed '^^

**Review by Orelsanpls — APPROVED** [review]
Let' go! good job

**Pull request merged**